lib/attrsets.nix

```diff
@@ -1094,6 +1094,44 @@ rec {
     attrs:
     map (name: f name attrs.${name}) (attrNames attrs);
 
+  /**
+    Call a function for each attribute in the given set and return
+    the concatenated results.
+
+    # Inputs
+
+    `f`
+
+    : A function, given an attribute's name and value, returns a list of new values.
+
+    `attrs`
+
+    : Attribute set to map over.
+
+    # Type
+
+    ```
+    # FIXME
+    concatMapAttrsToList :: (String -> a -> b) -> AttrSet -> [b]
+    ```
+
+    # Examples
+    :::{.example}
+    ## `lib.attrsets.concatMapAttrsToList` usage example
+
+    ```nix
+    concatMapAttrsToList (name: value: [ name (name + value) ])
+       { x = "a"; y = "b"; }
+    => [ "x" "xa" "y" "yb" ]
+    ```
+
+    :::
+  */
+  concatMapAttrsToList =
```

Comment:
Naming is hard. `concatMapAttrs` is taken by what should be `mergeMapAttrs`. I've also considered `mapAttrsToConcatList`.